networking/src/p2p/peer.rs

```diff
@@ -438,6 +445,24 @@ pub async fn bootstrap(
     let connection_message =
         ConnectionMessage::from_bytes(received_connection_message_bytes.content())?;
 
+    // create PublicKey from received bytes from remote peer
+    let peer_public_key = PublicKey::from_bytes(connection_message.public_key())?;
+
+    let connecting_to_self = peer_public_key == info.identity.public_key;
+    if connecting_to_self {
+        debug!(log, "Detected self connection");
+        // treat as if nack was received
+        return Err(PeerError::NackWithMotiveReceived {
+            nack_info: NackInfo::new(NackMotive::AlreadyConnected, &[]),
+        });
+    }
+
+    // make sure the peer performed enough crypto calculations
+    match check_proof_of_work(&received_connection_message_bytes.raw()[4..60], pow_target) {
```

Comment:
@akoptelov 
I think, this will be enought:
```
if let Err(e) = check_proof_of_work(...) {
   Err(e) => return Err(PeerError::PowError(e)),
}
```